fearless/Modules/Staking/StakingPayoutConfirmation/StakingPayoutConfirmationInteractor.swift

```diff
@@ -1,7 +1,126 @@
-import UIKit
+import Foundation
+import SoraKeystore
+import CommonWallet
+import RobinHood
+import IrohaCrypto
 
 final class StakingPayoutConfirmationInteractor {
+    let extrinsicService: ExtrinsicServiceProtocol
+    let signer: SigningWrapperProtocol
+    let balanceProvider: AnyDataProvider<DecodedAccountInfo>
+    let settings: SettingsManagerProtocol
+    let payouts: [PayoutInfo]
+
     weak var presenter: StakingPayoutConfirmationInteractorOutputProtocol!
+
+    init(
+        extrinsicService: ExtrinsicServiceProtocol,
+        signer: SigningWrapperProtocol,
+        balanceProvider: AnyDataProvider<DecodedAccountInfo>,
+        settings: SettingsManagerProtocol,
+        payouts: [PayoutInfo]
+    ) {
+        self.extrinsicService = extrinsicService
+        self.signer = signer
+        self.balanceProvider = balanceProvider
+        self.settings = settings
+        self.payouts = payouts
+    }
+
+    // MARK: - Private functions
+
+    private func createExtrinsicBuilderClosure() -> ExtrinsicBuilderClosure? {
+        let addressFactory = SS58AddressFactory()
+        let callFactory = SubstrateCallFactory()
+        let chain = settings.selectedConnection.type.chain
+
+        let closure: ExtrinsicBuilderClosure = { builder in
+            try self.payouts.forEach { payout in
+                let address = try addressFactory
+                    .addressFromAccountId(data: payout.validator, type: chain.addressType)
+
+                let payoutCall = try callFactory.payout(
+                    validatorAddress: address,
```

Comment:
I suggest to just to make factory to accept AccountId instead of the address

Reply:
Valid point